source/slang/lower-to-ir.cpp

```diff
@@ -4765,6 +4794,37 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
                 // the setter body.
             }
 
+            {
+                
+                auto attr = decl->FindModifier<PatchConstantFuncAttribute>();
+
+                // I needed to test for patchConstantFuncDecl here
+                // because it is only set if validateEntryPoint is called with Hull as the required stage
+                // If I just build domain shader, and then the attribute exists, but patchConstantFuncDecl is not set
+                // and thus leads to a crash.
+                if (attr && attr->patchConstantFuncDecl)
+                {
+                    // We need to lower the function
+                    FuncDecl* patchConstantFunc = attr->patchConstantFuncDecl;
+                    assert(patchConstantFunc);
+
+                    // Lower the patch constant function
+                    lowerFuncDecl(patchConstantFunc);
```

Comment:
I'm not sure that it is necessary or even correct to call `lowerFuncDecl` here. The call to `ensureDecl` *should* make sure the function gets lowered if needed on-demand.

Reply:
I've just confirmed that the test still passes with this line removed, so I submitted a change to eliminate it.